_posts/2020-02-03-apache-beam-how-beam-runs-on-top-of-flink.md

```diff
@@ -0,0 +1,159 @@
+
+One of the most exciting developments in the Beam technology is the framework’s support for multiple programming languages including Java, Python, Go, Scala and SQL. Essentially, developers can write their applications in a programming language of their choice. Beam, with the help of the Runners, translates the program to one of the execution engines, as shown in the diagram below.
+
+<center>
+<img src="{{ site.baseurl }}/img/blog/flink-runner-beam-beam-vision.png" width="600px" alt="The vision of Apache Beam"/>
+</center>
+
+
+# Reasons to use Beam with Flink
+
+Why would you want to use Beam with Flink instead of directly using Flink? Ultimately, Beam and Flink complement each other and provide additional value to the user. The main reasons for using Beam with Flink are the following: 
+
+* Beam provides a unified API for both batch and streaming scenarios.
+* Beam comes with native support for different programming languages, like Python or Go with all their libraries like Numpy, Pandas, Tensorflow, or TFX.
+* You get the power of Apache Flink like its exactly-once semantics, strong memory management and robustness.
+* Beam programs run on your existing Flink infrastructure or infrastructure for other supported Runners, like Spark or Google Cloud Dataflow. 
+* You get additional features like side inputs and cross-language pipelines that are not supported natively in Flink but only supported when using Beam with Flink. 
+
+
+# The Flink Runner in Beam
+
+The Flink Runner in Beam translates Beam pipelines into Flink jobs. The translation can be parameterized using Beam's pipeline options which are parameters for settings like configuring the job name, parallelism, checkpointing, or metrics reporting.
+
+If you are familiar with a DataSet or a DataStream, you will have no problems understanding what a PCollection is. PCollection stands for parallel collection in Beam and is exactly what DataSet/DataStream would be in Flink. Due to Beam's unified API we only have one type of results of transformation: PCollection.
+
+Beam pipelines are composed of transforms. Transforms are like operators in Flink and come in two flavors: primitive and composite transforms. The beauty of all this is that Beam only comes with a small set of primitive transforms which are:
+- `Source` (for loading data)
```

Comment:
You need to add a blank line in front of ordered and unordered listings, otherwise they won't be correctly rendered by Jekyll.